misc/determine-version.py

```diff
@@ -139,10 +141,10 @@ def version_cmp(a, b):
 # Find the full and abbreviated SHAs of the commit
 git = subprocess.Popen(["git", "rev-list", "-1", REV],
                        stdout=subprocess.PIPE)
-full_rev = git.stdout.readlines()[0].strip()
+full_rev = git.stdout.readlines()[0].strip().decode()
```

Comment:
I think `decode()` should be done before `strip()`. Hopefully, *UTF-8* is designed in a way that there can be no difference when changing the order here, but the golden rule is to convert the data ASAP and further process it.

Reply:
I just added a separate commit for this, pushed.